Add unit testing

IvanLinnik opened the pull request:
Is it write naming for test classes and methods?

IvanLinnik committed:
add converters test classes
rename converter method from "map" to "convert"

IvanLinnik committed:
rename test method
remove incorrect method from some indus

IvanLinnik committed:
repair alignment

IvanLinnik committed:
add actual and expected episode comparison

mmarusich commented on src/test/java/io/skai/okta/internshipstreamingserviceparserxml/services/converter/EpisodeAdditionalInfoConverterTest.java:
```diff
@@ -0,0 +1,54 @@
+                                        .runtime("62 min")
+                                        .build();
+
+        additionalInfo = converter.convert(9379992, item);
```
you should run convert in test instead of setUp method, before hook used to init some stuff

mmarusich commented on src/test/java/io/skai/okta/internshipstreamingserviceparserxml/services/converter/EpisodeAdditionalInfoConverterTest.java:
```diff
@@ -0,0 +1,54 @@
+
+    @BeforeEach
+    void setUp() {
+        EpisodeAdditionalInfoConverter converter = new EpisodeAdditionalInfoConverter();
```
you can run it with Mockito and init converter via @injectmocks annotation

mmarusich commented on src/test/java/io/skai/okta/internshipstreamingserviceparserxml/services/converter/EpisodeConverterTest.java:
```diff
@@ -0,0 +1,69 @@
+                              .link("https://www.lostfilmtv1.site/mr/series/The_Witcher/season_2/episode_1/")
+                              .build();
+
+        episode = converter.convert(item);
```
same comments here

IvanLinnik committed:
refactoring according to review comments
add tests in EpisodeRepositoryTest.java
add EpisodeAdditionalInfoConverterTest.java
add delete method to repository to cleanup after tests

IvanLinnik commented:
holy cow, i even find bug in additionalInfoRepository)

mmarusich commented on src/main/java/io/skai/okta/internshipstreamingserviceparserxml/repository/impl/EpisodeAdditionalInfoRepository.java:
```diff
@@ -54,10 +54,10 @@ public void save(EpisodeAdditionalInfo episodeAdditionalInfo) {
     }
 
-    public EpisodeAdditionalInfo get(long id) {
+    public EpisodeAdditionalInfo get(long parent_id) {
```
rename method to findByParentId + follow Java naming conversions (parent_id ---> parentId)

mmarusich commented on src/test/java/io/skai/okta/internshipstreamingserviceparserxml/repository/EpisodeAdditionalInfoRepositoryTest.java:
```diff
@@ -0,0 +1,60 @@
+
+    @BeforeEach
+    void setUp() {
+        expectedInfo = EpisodeAdditionalInfo.builder()
```
you can init it in fields section, you don't have any dynamic info, there is no point to create same object again and again before each test

IvanLinnik replied:
if i used the same object can i put it in @BeforeAll?

mmarusich commented on src/test/java/io/skai/okta/internshipstreamingserviceparserxml/repository/EpisodeAdditionalInfoRepositoryTest.java:
```diff
@@ -0,0 +1,60 @@
+
+    @Test
+    public void testRepositoryShouldBeNotNull() {
+        assertNotNull(repository);
```
what this test verifies?

IvanLinnik replied:
have an issue before. repository was null.
this is also redundant, because if i do testEpisodeSuccessfulSaveAndGet i will have NPException there? i should remove this assertion or move it in testEpisodeSuccessfulSaveAndGet method?

mmarusich commented on src/test/java/io/skai/okta/internshipstreamingserviceparserxml/repository/EpisodeRepositoryTest.java:
```diff
@@ -0,0 +1,65 @@
+
+    @Test
+    public void testRepositoryShouldBeNotNull() {
+        assertNotNull(repository);
```
same Q

mmarusich commented on src/test/java/io/skai/okta/internshipstreamingserviceparserxml/services/converter/EpisodeAdditionalInfoConverterTest.java:
```diff
@@ -0,0 +1,60 @@
+
+    @Test
+    void testConvertAdditionalInfoShouldBeNotNull() {
+        assertNotNull(converter.convert(9379992, item));
```
this test is redundant as you have testConvertAdditionalInfoShouldBeSuccessful which checks the same thing + actual result

IvanLinnik replied:
i should remove this assertion fully or move it in testConvertAdditionalInfoShouldBeSuccessful()?

mmarusich commented on src/test/java/io/skai/okta/internshipstreamingserviceparserxml/services/converter/EpisodeConverterTest.java:
```diff
@@ -0,0 +1,73 @@
+
+    @Test
+    void testEpisodeShouldBeNotNull() {
+        assertNotNull(converter.convert(item));
```
same thing

IvanLinnik commented:
if i run tests with coverage (using intellij idea context menu) it shows me real situation with coverage? or i must use jacoco or something else?

IvanLinnik committed:
refactoring according to review comments (how i understand :) )
add XmlParserLostFilmTest.java

IvanLinnik commented on docker-compose.yml:
```diff
@@ -35,3 +35,9 @@ services:
       KAFKA_TOOLS_LOG4J_LOGLEVEL: ERROR
     logging:
       driver: none
+  wiremock:
+    container_name: app_mock
+    ports:
+      - "8089:8089"
+    image: "wiremock/wiremock"
+    command: --port 8089
```
after yesterday's commit i understand that this part isn't needed for unit testing. I have a conflict with ports.

IvanLinnik committed:
remove unnecessary wiremock docker container
add EpisodesControllerTest.java, LostFilmServiceTest.java

mmarusich commented on src/test/java/io/skai/okta/internshipstreamingserviceparserxml/services/impl/LostFilmServiceTest.java:
```diff
@@ -0,0 +1,84 @@
+        verify(repository, times(2)).saveOrUpdate(episode);
+        verify(producer, times(2)).produce(eq("omdb"), eq(episode));
+        /*
+        mockito couldn't check invocation of this methods in "saveOrUpdate()"?
```
no because repo is mocked and this is correct, repository implementations should be verified in repo test class not in service

mmarusich commented on build.gradle:
```diff
@@ -42,15 +42,13 @@ dependencies {
     testImplementation 'org.springframework.kafka:spring-kafka-test'
-
-
+    testImplementation 'org.springframework.cloud:spring-cloud-starter-contract-stub-runner'
```
why you need stub runner?

IvanLinnik replied:
its for XmlParserLostFilmTest

mmarusich reviewed (approved):
LGTM

IvanLinnik committed:
cleanup
